src/Symfony/Component/Translation/Bridge/PoEditor/PoEditorProvider.php

```diff
@@ -0,0 +1,204 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Bridge\PoEditor;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Translation\Exception\ProviderException;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\Provider\ProviderInterface;
+use Symfony\Component\Translation\TranslatorBag;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ *
+ * @experimental in 5.3
+ *
+ * In PoEditor:
+ *  * Terms refer to Symfony's translation keys;
+ *  * Translations refer to Symfony's translated messages;
+ *  * Tags refer to Symfony's translation domains
+ *
+ * PoEditor's API always returns 200 status code, even in case of failure.
+ */
+final class PoEditorProvider implements ProviderInterface
+{
+    private $apiKey;
+    private $projectId;
+    private $client;
+    private $loader;
+    private $logger;
+    private $defaultLocale;
+    private $endpoint;
+
+    public function __construct(string $apiKey, string $projectId, HttpClientInterface $client, LoaderInterface $loader, LoggerInterface $logger, string $defaultLocale, string $endpoint)
+    {
+        $this->apiKey = $apiKey;
+        $this->projectId = $projectId;
+        $this->client = $client;
+        $this->loader = $loader;
+        $this->logger = $logger;
+        $this->defaultLocale = $defaultLocale;
+        $this->endpoint = $endpoint;
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('%s://%s', PoEditorProviderFactory::SCHEME, $this->endpoint);
+    }
+
+    public function write(TranslatorBagInterface $translatorBag): void
+    {
+        $defaultCatalogue = $translatorBag->getCatalogue($this->defaultLocale);
+
+        if (!$defaultCatalogue) {
+            $defaultCatalogue = $translatorBag->getCatalogues()[0];
+        }
+
+        $terms = $translations = [];
+        foreach ($defaultCatalogue->all() as $domain => $messages) {
+            foreach ($messages as $id => $message) {
+                $terms[] = [
+                    'term' => $id,
+                    'reference' => $id,
+                    'tags' => [$domain],
+                ];
+            }
+        }
+        $this->addTerms($terms);
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            $locale = $catalogue->getLocale();
+            foreach ($catalogue->all() as $messages) {
+                foreach ($messages as $id => $message) {
+                    $translations[$locale][] = [
+                        'term' => $id,
+                        'translation' => [
+                            'content' => $message,
+                        ],
+                    ];
+                }
+            }
+            $this->addTranslations($translations[$locale], $locale);
```

Comment:
This method currently sends an HTPP request and checks the response internally. This prevents benefiting from the built-in concurrency of symfony/http-client for the loop over catalogues.